app/controllers/v1/profiles_controller.rb

```diff
@@ -93,7 +100,9 @@ def parent_profile
     def business_objective
       @business_objective ||= Pundit.policy_scope(
         current_user, BusinessObjective
-      ).from_title(resource_attributes&.dig(:business_objective))
+      ).from_title(resource_attributes&.dig(:business_objective)) do |new_bo|
+        audit_bo_creation(new_bo)
+      end
```

Comment:
There's this cool [`ActiveModel::Dirty module`](https://api.rubyonrails.org/classes/ActiveModel/Dirty.html) we can leverage:

```suggestion
      ).from_title(resource_attributes&.dig(:business_objective))
      audit_bo_creation(@business_objective) if @business_objective.previous_changes.any?
```

Reply:
Doesn't `previous_changes` have different semantics? It has to trust that the `from_title` does not do any update to an existing bo.

Reply:
There's currently no way for `from_title` to update an existing BO record.